File: problems/bc628921-4faf-4066-b017-1dca7d2e63e1/python/checker.py
import io
from contextlib import redirect_stdout

def run_checker(testcases, solution):
    # results:
    # - exception
    # - isPassed
    # - failedTestcases:
    #      - id
    #      - input
    #      - expected
    #      - got
    #      - stdOut
    results = {
        "isPassed": True,
        "failedTestcases": [],
    }

    for testcase in testcases:
        testcase_input = testcase["input"]
        expected = testcase["expected"]
        result = {}

        try:
            stdout = io.StringIO()
            with redirect_stdout(stdout):
                got = solution.toLower(testcase_input)

            if got != expected:
                result["id"] = testcase["id"]
                result["input"] = testcase_input
                result["expected"] = expected
                result["got"] = got
                result["stdout"] = stdout.getvalue()

                results["isPassed"] = False
                results["failedTestcases"].append(result)
        except Exception as e:
            results["exception"] = f"{e}"
            results["isPassed"] = False
            break

    return results

File: problems/80f40615-0502-4b2d-a383-27f315fd013b/go/checker.go
package main

import "log"

func runChecker(testcases []*Testcase) *RunResult {
	runResult := &RunResult{
		IsPassed: true,
	}

	for _, testcase := range testcases {
		input := testcase.Input
		expected := testcase.Expected
		gotValue, stdout, err := captureStdout(longestSubstringLength, input)
		if err != nil {
			log.Fatal(err)
		}

		got := int(gotValue[0].Int())

		if expected != got {

			runResult.FailedTestcases = append(runResult.FailedTestcases, &FailedTestcase{
				Id:       testcase.Id,
				Input:    testcase.Input,
				Expected: testcase.Expected,
				Got:      got,
				StdOut:   stdout,
			})
			runResult.IsPassed = false
		}
	}

	return runResult
}
